Sources/ShopifyCheckoutSheetKit/CheckoutWebView.swift

```diff
@@ -151,14 +151,23 @@ extension CheckoutWebView: WKScriptMessageHandler {
 	func userContentController(_ controller: WKUserContentController, didReceive message: WKScriptMessage) {
 		do {
 			switch try CheckoutBridge.decode(message) {
+			/// Completed event
 			case let .checkoutComplete(checkoutCompletedEvent):
-				CheckoutWebView.cache = nil
 				viewDelegate?.checkoutViewDidCompleteCheckout(event: checkoutCompletedEvent)
-			case .checkoutUnavailable:
+			/// Error: Checkout unavailable
+			case .checkoutUnavailable(let message):
+				viewDelegate?.checkoutViewDidFailWithError(error: .checkoutUnavailable(message: message ?? "Checkout unavailable."))
+			/// Error: Storefront not configured properly
+			case .storefrontConfigurationError(let message):
+				viewDelegate?.checkoutViewDidFailWithError(error: .checkoutUnavailable(message: message ?? "Storefront was not configured properly."))
```

Comment:
@kiftio I'm keeping the concept of `.storefrontConfigurationError` internally but exposing a `.checkoutUnavailable` error